src/tasks/SwiftExecution.ts

```diff
@@ -0,0 +1,46 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the VSCode Swift open source project
+//
+// Copyright (c) 2024 the VSCode Swift project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of VSCode Swift project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import * as vscode from "vscode";
+import { SwiftProcess } from "./SwiftProcess";
+import { SwiftPseudoterminal } from "./SwiftPseudoterminal";
+
+export interface SwiftExecutionOptions extends vscode.ProcessExecutionOptions {
+    presentation?: vscode.TaskPresentationOptions;
+}
+
+export class SwiftExecution extends vscode.CustomExecution {
+    private swiftProcess?: SwiftProcess;
+
+    constructor(
+        public readonly command: string,
+        public readonly args: string[],
+        public readonly options: SwiftExecutionOptions
+    ) {
```

Comment:
Are you thinking to make testing easier, i.e. can pass a stub?

Reply:
Just for ease of use. As soon as this is available I have a use for it. Currently `swift package unedited` is run using execFile because I need to parse the output if it fails. I'd prefer to push this to a task, like all the other swift operations.

Reply:
So you would be able to pass in your own event listener to listen for output. I'm not sure we want to let overwriting the event itself. It may be more clear in the subsequent patches for diagnostics but this would be how you could accomplish:
```
const execution = task.execution;
if (!(execution && execution instanceof SwiftExecution)) {
      return;
}
const swiftExecution = execution as SwiftExecution;
swiftExecution.onDidWrite(data => {
     // do something with data
});